contracts/carrot-app/src/state.rs

```diff
@@ -1,120 +1,27 @@
-        ensure!(
-            self.reward <= self.min_gas_balance,
-            AppError::RewardConfigError(
-                "reward should be lower or equal to the min_gas_balance".to_owned()
-            )
-        );
-        ensure!(
-            self.max_gas_balance > self.min_gas_balance,
-            AppError::RewardConfigError(
-                "max_gas_balance has to be bigger than min_gas_balance".to_owned()
-            )
-        );
-
-        // Check swap asset has pairing into gas asset
-        DexAssetPairing::new(self.gas_asset.clone(), self.swap_asset.clone(), dex_name)
-            .resolve(&deps.querier, ans_host)?;
-
-        Ok(())
-    }
-}
-
-#[cw_serde]
-pub struct PoolConfig {
-    pub pool_id: u64,
-    pub token0: String,
-    pub token1: String,
-    pub asset0: AssetEntry,
-    pub asset1: AssetEntry,
-}
+// TEMP VARIABLES FOR DEPOSITING INTO ONE STRATEGY
+pub const TEMP_CURRENT_COIN: Item<Coin> = Item::new("temp_current_coins");
+pub const TEMP_EXPECTED_SWAP_COIN: Item<Uint128> = Item::new("temp_expected_swap_coin");
+pub const TEMP_DEPOSIT_COINS: Item<Vec<Coin>> = Item::new("temp_deposit_coins");
+pub const TEMP_CURRENT_YIELD: Item<usize> = Item::new("temp_current_yield_type");
 
-pub fn assert_contract(info: &MessageInfo, env: &Env) -> AppResult<()> {
-    if info.sender == env.contract.address {
-        Ok(())
-    } else {
-        Err(AppError::Unauthorized {})
-    }
-}
+pub type Config = ConfigBase<Checked>;
+pub type ConfigUnchecked = ConfigBase<Unchecked>;
 
 #[cw_serde]
-pub struct Position {
-    pub owner: Addr,
-    pub position_id: u64,
-    pub last_compound: Timestamp,
-}
-
-pub fn get_position(deps: Deps) -> AppResult<Position> {
-    POSITION
-        .load(deps.storage)
-        .map_err(|_| AppError::NoPosition {})
-}
-
-pub fn get_osmosis_position(deps: Deps) -> AppResult<FullPositionBreakdown> {
-    let position = get_position(deps)?;
-
-    ConcentratedliquidityQuerier::new(&deps.querier)
-        .position_by_id(position.position_id)
-        .map_err(|e| AppError::UnableToQueryPosition(position.position_id, e))?
-        .position
-        .ok_or(AppError::NoPosition {})
-}
-
-pub fn get_position_status(
-    storage: &dyn Storage,
-    env: &Env,
-    cooldown_seconds: u64,
-) -> AppResult<CompoundStatus> {
-    let position = POSITION.may_load(storage)?;
-    let status = match position {
-        Some(position) => {
-            let ready_on = position.last_compound.plus_seconds(cooldown_seconds);
-            if env.block.time >= ready_on {
-                CompoundStatus::Ready {}
-            } else {
-                CompoundStatus::Cooldown((env.block.time.seconds() - ready_on.seconds()).into())
-            }
-        }
-        None => CompoundStatus::NoPosition {},
-    };
-    Ok(status)
+pub struct ConfigBase<T> {
+    pub balance_strategy: BalanceStrategyBase<T>,
+    pub autocompound_config: AutocompoundConfigBase<T>,
+    pub dex: String,
```

Comment:
Do we need to save it in state if we just use const variable, or is it for future integration with other dexes? Why not `Vec<dex>` if so?

Reply:
The dex here doesn't matter at all. It doesn't have to be a const variable. We don't use a CONST variable inside the contract at all.